be/src/agent/topic_subscriber.cpp

```diff
@@ -38,13 +38,13 @@ TopicSubscriber::~TopicSubscriber() {
 
 void TopicSubscriber::register_listener(TTopicType::type topic_type, TopicListener* listener) {
     // Unique lock here to prevent access to listeners
-    WriteLock lock(_listener_mtx);
+    std::lock_guard<std::shared_mutex> lock(_listener_mtx);
     this->_registered_listeners[topic_type].push_back(listener);
 }
 
 void TopicSubscriber::handle_updates(const TAgentPublishRequest& agent_publish_request) {
     // Shared lock here in order to avoid updates in listeners' map
-    ReadLock lock(_listener_mtx);
+    std::shared_lock<std::shared_mutex> lock(_listener_mtx);
```

Comment:
Using std::shared_lock lock(_listener_mtx); instead of std::shared_lock<std::shared_mutex> lock(_listener_mtx);